keps/sig-network/0006-20180314-coredns-GA-proposal.md

```diff
@@ -0,0 +1,125 @@
+Some of the most used deployment tools have been upgraded by the CoreDNS team, in cooperation of the owners of these tools, to be able to deploy CoreDNS:
+* kubeadm
+* kube-up
+* minikube
+* kops
+
+For other tools, each maintainer would have to add the upgrade to CoreDNS.
+
+### Use Cases
+
+* CoreDNS supports all functionality of kube-dns and also addresses [several use-cases kube-dns lacks](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/network/coredns.md#use-cases). Some of the Use Cases are as follows: 
+    * Supporting [Autopath](https://coredns.io/plugins/autopath/), which reduces the high query load caused by the long DNS search path in Kubernetes.
+    * Making an alias for an external name [#39792](https://github.com/kubernetes/kubernetes/issues/39792)
+    
+* By default, the user experience would be unchanged. For more advanced uses, existing users would need to modify the ConfigMap that contains the CoreDNS configuration file.
+* Since CoreDNS has more supporting features than kube-dns, there will be no path to retain the CoreDNS configuration in case a user wants to switch to kube-dns.
+
+#### Configuring CoreDNS
+
+The CoreDNS configuration file is called a `Corefile` and syntactically is the same as a [Caddyfile](https://caddyserver.com/docs/caddyfile). The file consists of multiple stanzas called _server blocks_.
+Each of these represents a set of zones for which that server block should respond, along with the list of plugins to apply to a given request. More details on this can be found in the 
+[Corefile Explained](https://coredns.io/2017/07/23/corefile-explained/) and [How Queries Are Processed](https://coredns.io/2017/06/08/how-queries-are-processed-in-coredns/) blog entries.
+
+The following can be expected when CoreDNS is graduated to GA.
+
+#### Kubeadm
+
+* The CoreDNS feature-gates flag will be set to default and marked as GA.
+* On a fresh install of Kubernetes via kubeadm by specifying the `kubeadm init` command, CoreDNS will be the default installed DNS add-on.
+* In case a user wants to install kube-dns instead of CoreDNS, they have to set the feature-gate of CoreDNS to false. `--feature-gates=CoreDNS=false`
+* For users upgrading Kubernetes via kubeadm, it will install CoreDNS whether the user had kube-dns or CoreDNS in a previous kubernetes version (Unless the CoreDNS feature-gate is disabled).
+* The configmap of a previously installed kube-dns will be automatically translated to the equivalent CoreDNS configmap.
+
+#### Kube-up
+
+* CoreDNS set as the default DNS service in place of kube-dns.
+
+#### Minikube
+
+* CoreDNS to be enabled by default in the add-on manager, with kube-dns disabled by default.
+ 
+## Unresolved questions
+
+* Fate of kube-dns in future releases, i.e. deprecation path.
+
+## Graduation Criteria
+
```

Comment:
@rramkumar1 Just to clarify:
I submit a PR to merge now making CoreDNS default in kubeadm, kube-up... to see if the tests are "passing" and if we're satisfied with the result, we keep the PR, else we revert.

And what is the right number of runs to determine it's a success?

Reply:
@rramkumar1 : why would we need to specify "amount of test runs" ? I would think that one is enough no ? as long as all test are OK with CoreDNS then CoreDNS is acceptable.

Are you worry about kind of flaking tests ? or other commits that would interfere and failed some of the  "conformance" tests (independently of DNS service) ?

To prevent to have a wrong view do to other errors )not linked to the DNS service), we could propose to have one of the "conformance" + "DNS" tests running with kube-dns and one with CoreDNS. so we can validate we have the same level of passing tests.

Reply:
For e2e test, I would propose:

- all tests filtered by "--ginkgo.skip=\\[Slow\\]|\\[Serial\\]|\\[Disruptive\\]|\\[Flaky\\]|\\[Feature:.+\\]"
(the regular tests "xxx-kubernetes-e2e-gce").

As the success of each test depends of the code itself (not only the DNS Service), we would verify that CoreDNS is equivalent to kube-dns in regards of the passed e2e tests. That means validate
"ci-kubernetes-e2e-gce-gci-ci-master" against the exact same test "ci-kubernetes-e2e-gce-gci-ci-master-kube-dns" with kube-dns deployed. 

They are supposed to be always the same, or at least none of the success tests with kube-dns is failing with coredns.